### ISSUE TITLE
Reference sequence basename and comment

### ISSUE THREAD

**Issue by ocxtal** [jason-weirather/AlignQC#2]
Hi, Jason

Running AlignQC with reference sequence containing comments (separated by a space after sequence basename) fails with a keyerror.

For example, reference sequence with the following name:
```
>gi|556503834|ref|NC_000913.3| Escherichia coli str. K-12 substr. MG1655, complete genome
```

generates following error:
```
Traceback (most recent call last):
  File ~/AlignQC/bin/alignqc, line 44, in <module>
    main()
  File ~/AlignQC/bin/alignqc, line 24, in main
    analyze.external_cmd( .join(operable_argv),version=version)
  File ~/AlignQC/utilities/analyze.py, line 77, in external_cmd
    main(args)
  File ~/AlignQC/utilities/analyze.py, line 44, in main
    prepare_all_data.external(args)
  File ~/AlignQC/utilities/prepare_all_data.py, line 569, in external
    main(args)
  File ~/AlignQC/utilities/prepare_all_data.py, line 65, in main
    make_data_bam_reference(args)
  File ~/AlignQC/utilities/prepare_all_data.py, line 300, in make_data_bam_reference
    bam_to_context_error_plot.external_cmd(cmd)
  File ~/AlignQC/utilities/bam_to_context_error_plot.py, line 140, in external_cmd
    main(args)
  File ~/AlignQC/utilities/bam_to_context_error_plot.py, line 48, in main
    epf.add_alignment(e)
  File ~/AlignQC/pylib/Bio/Errors.py, line 46, in add_alignment
    ae = AlignmentErrors(align)
  File ~/AlignQC/pylib/Bio/Errors.py, line 453, in __init__
    astrings = self._alignment.get_alignment_strings(min_intron_size=self._min_intron_size)
  File ~/AlignQC/pylib/Bio/Align.py, line 122, in get_alignment_strings
    tdone += textra+ref[t.chr][t.start-1:t.end].upper()
  File ~/AlignQC/pylib/Bio/Format/Fasta.py, line 74, in __getitem__
    return self._seqs[key]
KeyError: 'gi|556503834|ref|NC_000913.3|'
```

The reason seems that the reference names are handled with comments but alignment tools include only the basenames in the output sam file (BWA-MEM and GraphMap are tested, my mapping tool minialign is also designed so). How about changing AlignQC to handle reference sequences only with their basenames? And is there any problems (like name conflict) on this?

And here is my quick fix of this based on master: https://github.com/ocxtal/AlignQC/tree/basename

Thanks,

Hajime

**Comment by jason-weirather** [comment]
Hi Hajime,

Thank you very much for pointing this out.  I think your recommendation to change AlignQC to only handle the basename of the reference sequence is the right thing to do.  I'll try to get this fix into the next update to AlignQC.  The next update will be a fairly substantial one because there are some serious memory bubbles and performance issues that crop up once you get into a very large number of reads that I will be improving upon.  Hopefully I can get these fixes out in the next week and a half.

**Comment by ocxtal** [comment]
Hi Jason

Thank you very much for your detailed explanation. I hope the next release will be a good one. 😆😆😆

Hajime

**Comment by jason-weirather** [comment]
Fix for this should be in now with v1.2 just released.  If you have any more issues, or any problem with compatibility with your aligner please let me know.  I'd like to make sure its supported.

**Comment by ocxtal** [comment]
Thanks!! 👍👍👍👍 I'll try it.

Hajime
